app/src/main/java/unipd/se18/ocrcamera/AllergensManager.java

```diff
@@ -0,0 +1,254 @@
+package unipd.se18.ocrcamera;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.opencsv.CSVReader;
+
+import unipd.se18.ocrcamera.inci.Ingredient;
+
+
+
+/**
+ * Class that contains the methods to read the allergens from a file, create an allergen list,
+ * find the allergens in the recognized ingredients from the OCR.
+ */
+class AllergensManager {
+
+    private final int COMMON_NAME = 0;
+    private final int INCI_NAMES = 1;
+
+    private ArrayList<Allergen> allergensList = new ArrayList<>();
+
+    private Context context;
+    private SharedPreferences sp = null;
+
+    /**
+     * constructor, accesses the sharedPreferences file and reads the allergens database from res
+     * @param cntxt context of the app/calling activity
+     * @author Cervo Nicolò
+     */
+    AllergensManager(Context cntxt) {
+
+        context = cntxt;
+        String selectedAllergensSP = cntxt.getString(R.string.selected_allergens_sharedPref);
+        sp = context.getSharedPreferences(selectedAllergensSP, Context.MODE_PRIVATE);
```

Comment:
I don't understand why the Context object is referred both as "context" and "cntxt" in the lines above. Is there a particular reson?

Reply:
I could have used this.context, you are right, will fix

Reply:
Sorry for not having been clear enough: I referred to the fact that getString is called on "cntxt" while getSharedPreferences is on "context" instead

Reply:
You are right, I didn't even realize that, there is no reason, will fix